framework/physics/physics_material/multi_group_xs/single_state_mgxs.cc

```diff
@@ -887,225 +818,216 @@ SingleStateMGXS::MakeFromChiXSFile(const std::string& file_name)
-      nu_sigma_f_.assign(num_groups_, 0.0);
-      for (unsigned int g = 0; g < num_groups_; ++g)
-        nu_sigma_f_[g] = nu[g] * sigma_f_[g];
+      // Compute total production cross section
+      nu_sigma_f_ = sigma_f_;
+      for (size_t g = 0; g < num_groups_; ++g)
+        nu_sigma_f_[g] *= nu[g];
 
-      // compute prompt production cross section
+      // Compute prompt production cross section
       if (not nu_prompt.empty())
       {
-        nu_prompt_sigma_f_.assign(num_groups_, 0.0);
-        for (unsigned int g = 0; g < num_groups_; ++g)
-          nu_prompt_sigma_f_[g] = nu_prompt[g] * sigma_f_[g];
+        nu_prompt_sigma_f_ = sigma_f_;
+        for (size_t g = 0; g < num_groups_; ++g)
+          nu_prompt_sigma_f_[g] *= nu_prompt[g];
       }
 
-      // compute delayed production cross section
+      // Compute delayed production cross section
       if (not nu_delayed.empty())
       {
-        nu_delayed_sigma_f_.assign(num_groups_, 0.0);
-        for (unsigned int g = 0; g < num_groups_; ++g)
-          nu_delayed_sigma_f_[g] = nu_delayed[g] * sigma_f_[g];
+        nu_delayed_sigma_f_ = sigma_f_;
+        for (size_t g = 0; g < num_groups_; ++g)
+          nu_delayed_sigma_f_[g] *= nu_delayed[g];
       }
 
-      // compute production matrix
-      const auto chi_ = not chi_prompt.empty() ? chi_prompt : chi;
+      // Compute production matrix
+      const auto fis_spec = not chi_prompt.empty() ? chi_prompt : chi;
       const auto nu_sigma_f = not nu_prompt.empty() ? nu_prompt_sigma_f_ : nu_sigma_f_;
 
-      for (unsigned int g = 0; g < num_groups_; ++g)
-      {
-        std::vector<double> prod;
-        for (unsigned int gp = 0.0; gp < num_groups_; ++gp)
-          prod.push_back(chi_[g] * nu_sigma_f[gp]);
-        production_matrix_.push_back(prod);
-      }
+      production_matrix_.resize(num_groups_);
+      for (size_t g = 0; g < num_groups_; ++g)
+        for (size_t gp = 0.0; gp < num_groups_; ++gp)
+          production_matrix_[g].push_back(fis_spec[g] * nu_sigma_f[gp]);
     } // if production_matrix empty
 
     else
     {
-      // TODO: Develop an implementation for multi-particle delayed
-      //       neutron data. The primary challenge in this is that
-      //       different precursor species exist for neutron-induced
-      //       fission than for photo-fission.
+      // TODO: Develop an implementation for multi-particle delayed neutron data.
+      //       The primary challenge in this is that different precursor species exist for
+      //       neutron-induced fission than for photo-fission.
 
-      // throw error if precursors are present
-      if (num_precursors_ > 0)
-        throw std::runtime_error("This setup has not been implemented.\n"
-                                 "Currently, when a production matrix is specified, "
-                                 "no delayed neutron precursors are allowed.");
+      // Not implemented error
```

Comment:
This comment seems incorrect - it is not implemented error

Reply:
Well this is obvious, haha. I didn't want to go create a macro for not implemented error so I just shortcut it. I will use a proper error.

Reply:
My point is that you correctly handled an error state - that's is not "not implemented" error IMO. So, just remove the comment, if you plan to add something more in future. This check would be replaced, then with more code.